src/flight_mode/src/gpio.rs

```diff
@@ -1,8 +1,149 @@
-pub fn init() {
-    println!("flight mode gpio init");
+
+//! GPIO access under Linux using the rust-embedded sysfs_gpio library 
+//! (https://github.com/rust-embedded/rust-sysfs-gpio)
+//!
+//! The methods exposed by this module are centered around
+//! the `MyPins` struct.
+
+use sysfs_gpio::{Pin, Direction};
+use std::collections::LinkedList;
+
+struct Node {
+    num: u64,
+    dir: Direction,
+    io: Pin,
+}
+pub struct MyPins {
+    pins: LinkedList<Node>,
+}
+
+/// Unexports any pins that were added to MyPins.
+impl Drop for MyPins {
+    fn drop(&mut self) {
+        for pin in &self.pins {
+            if let Err(err) = pin.io.unexport() {
+                println!("gpio pin {} unexport error: {}", pin.num, err);
+            }
+        }
+    }
+}
+
+
+impl MyPins {
+
+/// Create a new MyPins object. MyPins can contain multiple GPIO pins
+/// that are created with the `add_pin` method.
+pub fn new() -> MyPins {
+    MyPins {
+        pins: LinkedList::new(),
+    }
+}
+
+
+/// Used to add GPIO pins to a MyPins object. Sets the direction of 
+/// the pin and exports it.
+///
+/// Inputs:
+///
+///    `pin_number` -- the desired GPIO pin number
+///
+///    `direction` -- `"in"` configures a pin as an input, 
+///                   `"out"` configures a pin as an output, 
+///                   `"high"` configures pin as an output and sets its value to 1, and 
+///                   `"low"` configures pin as an output and sets its value to 0.
+///
+/// # Panics
+/// 1) User does not have root privileges.
+/// 2) Invalid 'direction' argument.
+/// 3) The system does not support the GPIO sysfs interface.
+/// 4) The requested GPIO is out of range and cannot be exported.
+/// 5) The requested GPIO is in use by the kernel and cannot
+///    be exported by use in userspace.
+pub fn add_pin(&mut self, pin_number: u64, direction: &str) {
```

Comment:
Jamie: Consider not keeping the pin direction ourselves, the pin object should have that data already.